packages/flutter_test/lib/src/binding.dart

```diff
@@ -498,11 +498,11 @@ abstract class TestWidgetsFlutterBinding extends BindingBase
       }
     );
     _parentZone = Zone.current;
+    asyncBarrier(); // When using AutomatedTestWidgetsFlutterBinding, this flushes the microtasks.
     final Zone testZone = _parentZone.fork(specification: errorHandlingZoneSpecification);
     testZone.runBinary(_runTestBody, testBody, invariantTester)
       .whenComplete(testCompletionHandler);
     timeout?.catchError(handleUncaughtError);
-    asyncBarrier(); // When using AutomatedTestWidgetsFlutterBinding, this flushes the microtasks.
     return testCompleter.future;
   }
```

Comment:
I don't understand this change. What does it even mean to have an async barrier before the test runs? The whole point here is to catch the case of people having guarded calls that are still open after the test has returned.

Maybe the line should be removed and moved to the end of runTest?

Reply:
done.